common/models/templates/src/trie-model.ts

```diff
@@ -161,6 +161,116 @@
+              //
+              // Note:  Things get nasty here if this is only sometimes true; in the future,
+              // we should compile-time enforce that this assumption is always true if possible.
+              if(entryNode.type == 'internal') {
+                let internalNode = entryNode;
+                for(let j = 0; j < entryNode.values.length; j++) {
+                  let prefix = this.prefix + entry + internalNode.values[j];
+                  yield {
+                    char: entry + entryNode.values[j],
+                    traversal: function() { return new TrieModel.Traversal(internalNode.children[internalNode.values[j]], prefix) }
+                  }
+                }
+              } else {
+                // Determine how much of the 'leaf' entry has no Trie nodes, emulate them.
+                let fullText = entryNode.entries[0].key;
+                entry = entry + fullText[this.prefix.length + 1]; // The other half of the SMP.
+                let prefix = this.prefix + entry;
+
+                yield {
+                  char: entry,
+                  traversal: function () {return new TrieModel.Traversal(entryNode, prefix)}
+                }
+              }
+            } else if(charCode == 0xFDD0) {
+              continue;
+            } else {
+              let prefix = this.prefix + entry;
+              yield {
+                char: entry,
+                traversal: function() { return new TrieModel.Traversal(entryNode, prefix)}
+              }
+            }
+          }
+
+          return;
+        } else { // type == 'leaf'
+          let prefix = this.prefix;
+
+          let children = root.entries.filter(function(entry) {
+            return entry.key != prefix && prefix.length < entry.key.length;
+          })
+
+          for(let i = 0; i < children.length; i++) {
+            let entry = children[i];
+            let key = entry.key;
+            let nodeKey = entry.key[prefix.length]
+            let charCode = nodeKey.charCodeAt(0);
+
+            if(charCode >= 0xD800 && charCode <= 0xDBFF) {
+              // Merge the other half of an SMP char in!
+              nodeKey = nodeKey + key[prefix.length+1];
+            }
+            yield {
+              char: nodeKey,
+              traversal: function() { return new TrieModel.Traversal(root, prefix + nodeKey)}
+            }
+          };
+          return;
+        }
+      }
+
+      get entries(): string[] {
+        if(this.root.type == 'leaf') {
+          let prefix = this.prefix;
+
+          let matches = this.root.entries.filter(function(entry) {
+            return entry.key == prefix;
+          })
+          if(matches.length > 0) {
+            return matches.map(function(value) { return value.content });
+          } else {
+            return undefined;
+          }
+        } else {
+          let matchingLeaf = this.root.children['\uFDD0'];
```

Comment:
TODO: extract the constant `\uFDD0` into a constant... maybe with a better name than this: https://github.com/keymanapp/keyman/blob/df283e6fd7e857835a67114953c2b009f74bd9f1/developer/js/source/lexical-model-compiler/build-trie.ts#L284